py_proxy/views.py

```diff
@@ -70,22 +74,26 @@ def content_type(request):
     )
 
     with requests.get(url, stream=True, allow_redirects=True) as rsp:
-        if rsp.headers.get("Content-Type") in ("application/x-pdf", "application/pdf",):
+        if rsp.headers.get("Content-Type") in ("application/x-pdf", "application/pdf"):
             return exc.HTTPFound(
                 request.route_url(
-                    "pdf", pdf_url=request.matchdict["url"], _query=request.params
-                )
+                    "pdf", pdf_url=request.matchdict["url"], _query=request.params,
+                ),
+                headers=_caching_headers(max_age=300),
             )
 
     via_url = request.registry.settings["legacy_via_url"]
-    url = _drop_from_url_begining("/", request.path_qs)
-    return exc.HTTPFound(f"{via_url}/{url}")
+    url = request.path_qs.lstrip("/")
 
+    return exc.HTTPFound(f"{via_url}/{url}", headers=_caching_headers(max_age=60))
 
-def _drop_from_url_begining(drop_chars, url):
-    """Drop drop_chars from begining of url."""
-    drop_before = len(drop_chars)
-    return url[drop_before:]
+
+def _caching_headers(max_age, stale_while_revalidate=86400):
+    # I tried using webob.CacheControl for this but it's total rubbish
+    header = (
+        f"public, max-age={max_age}, stale-while-revalidate={stale_while_revalidate}"
+    )
```

Comment:
I tried using `webob.CacheControl`, but that module is a total disaster. It will cheerfully create an invalid `Cache-Control` header if you send it even mildly the wrong thing. It's not worth it.